habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDependenciesMojo.java

```diff
@@ -0,0 +1,156 @@
+package org.technologybrewery.habushu;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.shared.model.fileset.FileSet;
+import org.apache.maven.shared.model.fileset.util.FileSetManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Stages the source files of a monorepo dependency
+ * to the target directory along with the source files of
+ * any transitive path-based dependencies.
+ */
+
+@Mojo(name = "containerize-dependencies", defaultPhase = LifecyclePhase.VALIDATE)
+public class ContainerizeDependenciesMojo extends AbstractHabushuMojo {
+
+    private static final Logger logger = LoggerFactory.getLogger(ContainerizeDependenciesMojo.class);
+
+    @Component
+    protected MavenSession session;
+
+    protected File anchorSourceDirectory;
+    protected File anchorOutputDirectory;
+    protected final String HABUSHU = "habushu";
+
+    /**
+     * Overriding to allow execution in non-habushu projects.
+     */
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        doExecute();
+    }
+
+    @Override
+    protected void doExecute() throws MojoExecutionException, MojoFailureException {
+        if (this.anchorSourceDirectory == null) {
+            this.anchorSourceDirectory = new File(session.getExecutionRootDirectory());
+        }
+
+        Set<MavenProject> habushuProjects = getHabushuProjects();
+        try {
+            copyRelevant(habushuProjects);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Copies the relevant source files by leveraging {@link FileSet}s to filter appropriately.
+     * @param habushuProjects corresponding projects of the pom's habushu-type dependencies
+     * @throws IOException
+     */
+    private void copyRelevant(Set<MavenProject> habushuProjects) throws IOException {
+        this.anchorOutputDirectory = new File(project.getBuild().getDirectory()
+                + "/venv-support/" + this.anchorSourceDirectory.getName());
+        Path srcRoot = this.anchorSourceDirectory.toPath();
+        Path destRoot = this.anchorOutputDirectory.toPath();
```

Comment:
A: If we're only using these `Path` objects, seems like we should just construct them directly instead of constructing a `File` first and converting to `Path`.

Reply:
addressed, see changes throughout the mojo and test steps

Reply:
Update: these were changed back to `File` objects because mojos cannot create `Path` objects via the `@Parameter()`. (This is a result of making the anchor attributes configurable mojo parameters per other feedback)